source/usb_hid.c

```diff
@@ -473,16 +714,25 @@ static int usb_hid_worker(void *arg)
 		return ret;
 	host_fd = ret;
 
-	ret = os_ioctl(host_fd, USBV5_IOCTL_GETVERSION, NULL, 0, ver, sizeof(ver));
-	if (ret < 0)
-		return ret;
-
-	/* We only support USBv5 for now */
-	if (ver[0] != 0x50001)
-		return IOS_EINVAL;
-
-	ret = os_ioctl_async(host_fd, USBV5_IOCTL_GETDEVICECHANGE, NULL, 0, device_change_devices,
+	ret = os_ioctl(host_fd, USBV4_IOCTL_GETVERSION, NULL, 0, NULL, 0);
+	if (ret == 0x40001) {
+		using_v4_api = true;
```

Comment:
Shouldn't this be a property of the specific USB controller, instead of being set globally?

Reply:
the API in use is down to the IOS version not the device, so theres no point in doing that per device.

Reply:
You are right, I thought that this was a property of the attached USB device.